.github/workflows/test-external.yml

```diff
@@ -67,35 +85,11 @@ jobs:
           ref: ${{ matrix.project.branch }}
           submodules: recursive
 
-      - name: Install Foundry
-        uses: foundry-rs/foundry-toolchain@v1
-
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.12"
-
-      - name: Install dependencies
-        run: python -m pip install --upgrade pip
-
-      - name: Install Halmos
-        run: python -m pip install -e ./halmos
-
-      - name: Install Vyper
-        if: ${{ matrix.project.dir == 'snekmate' }}
-        run: python -m pip install vyper
-
-      - name: Install Yices 2 SMT solver
-        run: |
-          wget https://github.com/SRI-CSL/yices2/releases/download/Yices-2.6.4/yices-2.6.4-x86_64-pc-linux-gnu.tar.gz
-          tar -xzvf yices-2.6.4-x86_64-pc-linux-gnu.tar.gz
-          sudo mv yices-2.6.4/bin/* /usr/local/bin/
-          sudo mv yices-2.6.4/lib/* /usr/local/lib/
-          sudo mv yices-2.6.4/include/* /usr/local/include/
-          rm -rf yices-2.6.4
+      - name: Print halmos version
+        run: docker run -v .:/workspace halmos-image --version
```

Comment:
the halmos version printed here is quite different from what we get locally.
```
Run docker run -v .:/workspace halmos-image --version
halmos 0.1.dev1+g81b7ae5
```
https://github.com/a16z/halmos/actions/runs/10014074545/job/27683005509

can we improve the dockerfile to contain the correct halmos version?

Reply:
the only difference I can see is that we install halmos via `uv pip install` in the Dockerfile instead of `pip install`

Reply:
oh and actually the other big difference is the sparse checkout we do in github actions, which explains the `0.1.dev1` part. I think it's fine, the git hash has the information we need really